### ISSUE TITLE
Incorrect interledger payment pointer added to my post

### ISSUE THREAD

**Issue by devcer** [forem/forem#18757]
**Describe the bug**

I have webmonetized my [dev.to](http://dev.to/) profile by adding a payment pointer to my profile. Thanks for building the feature. I have started receiving tips for my content as well. However I see I did not receive any micropayments from the past 5 days.

I have tipped myself using Coil to validate this and I did not receive the payment. I went through the code through `View Source` and I found that an incorrect payment pointer is being added to my post.

My profile is configured with payment pointer `$ilp.uphold.com/wypi4eKYYYJw` but I see the pointer `$ilp.uphold.com/24HhrUGG7ekn` being shown in the page.

People have been tipping in front of me but I failed to receive any payments. Can you please check and let me know what went wrong here?

**To Reproduce**

1. Go to 'https://dev.to/devcer/10-must-read-guides-for-open-payments-and-web-monetization-enthusiasts-c5a'
2. Right click and select `View page source`
3. Do `cmd+f` and search for monetization
4. Observe the payment pointer

**Expected behavior**

My profile is configured with payment pointer `$ilp.uphold.com/wypi4eKYYYJw` but I see the pointer `$ilp.uphold.com/24HhrUGG7ekn` being shown in the page.

**Screenshots**
<img width="1552" alt="Screenshot 2022-11-13 at 15 39 33" src="https://user-images.githubusercontent.com/3799600/202865216-65c181da-1db3-4268-a114-38e4116ba086.png">
<img width="1552" alt="Screenshot 2022-11-13 at 15 40 58" src="https://user-images.githubusercontent.com/3799600/202865220-165cefc3-bccf-45e2-b4b0-84e0a9809278.png">

**Desktop (please complete the following information):**

- OS, version:
- Browser, version:

**Smartphone (please complete the following information):**

- Device:
- OS, version:
- Browser, version:

**Additional context**

**Comment by fdocr** [comment]
Thank you for the report @devcer. I can tell we have logic in place to update the payment pointer on your post. If I lookup your link I can see it there:

<img width="1070" alt="Screen Shot 2022-11-28 at 11 16 15" src="https://user-images.githubusercontent.com/6045239/204340315-cef9b5ce-8d5e-42aa-8cbb-79727d95a166.png">

However, that other link you mention is also available within the page

<img width="1078" alt="Screen Shot 2022-11-28 at 11 17 00" src="https://user-images.githubusercontent.com/6045239/204340458-dcf0de6e-c335-4c66-aac4-b289edd69986.png">

If you take a closer look you can tell that the second one is only visible within an `iframe`. That `iframe` is there as part of the moderation panel. I wonder if this is the reason your payments haven't gone through, if by mistake the tips are going to that other link.

I believe this can be fixed by ensuring the `iframe` doesn't include the other payment pointer. This can be done in various ways but worth the effort IMO

**Comment by devcer** [comment]
Thank you @fdocr for checking this. Also, Let me try to do a payment myself and see if I receive the payment. it didn't go through last time. I didn't know that a pointer inside an `iframe` can receive payments as well.

**Comment by fdocr** [comment]
Yes, that would be great if you can test that payments work with your pointer @devcer. Our implementation hasn't changed for some time now and I'm not well versed on the tools used for how to test this (either in production/sandbox environments).

> I didn't know that a pointer inside an iframe can receive payments as well.

That's just a hypothesis from my end. I can assure you the correct pointer you shared (`$ilp.uphold.com/wypi4eKYYYJw`) is available in the `meta` tag of the site.

The reason you don't see it when doing a "view page source" (after a _right click_ on the page) is that post author pointers are loaded with javascript when the page loads, when "viewing page source" directly javascript doesn't execute. If you open the developer tools and inspect the page elements you'll find the correct payment pointer loaded for your article.

<img width="1049" alt="Screen Shot 2022-11-29 at 09 04 38" src="https://user-images.githubusercontent.com/6045239/204565499-57b53902-4a1c-4078-864f-5521a953c8d2.png">
